### PR TITLE
feat(unlock-app): Improve key metadata editor

### PR DESCRIPTION
**Pull request by searchableguy**
# Description

CleanShot 2022-12-22 at 20 48 15@2x
# Issues

CleanShot 2022-12-22 at 20 50 00@2x

CleanShot 2022-12-22 at 20 50 36@2x

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by searchableguy** [comment]
CleanShot 2022-12-22 at 17 57 52@2x

File: packages/unlock-js/src/web3Service.ts [hunk 1]
```diff
@@ -928,15 +929,20 @@ export default class Web3Service extends UnlockService {
...
+    network,
+  }: {
+    lockAddress: string
+    network: number
+  }) {
+    const lockContract = await this.getLockContract(
+      lockAddress,
+      this.providerForNetwork(network)
+    )
+    const tokenURI = await lockContract.tokenURI(1)
```
Comment (julien51) [inline packages/unlock-js/src/web3Service.ts:943]: Very good! Thanks!

File: unlock-app/src/components/interface/Picker/index.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  lockAddress?: string
+  keyId?: string
+}
+
+interface Props extends PickerState {
+  userAddress: string
+  onChange(state: PickerState): void
+}
+
+export function Picker({
```
[the remark this answers is not in the document]
Comment (julien51) [reply unlock-app/src/components/interface/Picker/index.tsx:20]: I understand the current picker does not include the key, but I think we could add it as a config prop?

[the remark this answers is not in the document]
Comment (searchableguy) [reply unlock-app/src/components/interface/Picker/index.tsx:20]: The changes I wanted to make and also add support for key would require rewriting that entirely so I created a generic picker which I will replace that one with. It did not support passing values and refetched thousands of locks on network change.

**Review by julien51 — CHANGES_REQUESTED** [review]
Please, let's re-use the lockPicker (and modify it!) so we do not introduce additional complexity in an already complicated app :)

File: unlock-app/src/components/interface/Picker/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (julien51) [reply unlock-app/src/components/interface/Picker/index.tsx:20]: Are you planning to replace the other Picker with this new one? If so, please add a comment on the other one so we have a clear way forward?

[the remark this answers is not in the document]
Comment (searchableguy) [reply unlock-app/src/components/interface/Picker/index.tsx:20]: Yes. I wanted to put this in UI library and replace from it but that would be out of scope for this PR. I will add a comment.

[the remark this answers is not in the document]
Comment (searchableguy) [reply unlock-app/src/components/interface/Picker/index.tsx:22]: Added

[the remark this answers is not in the document]
Comment (searchableguy) [reply unlock-app/src/components/interface/Picker/index.tsx:23]: Removed.

File: unlock-app/src/components/interface/Picker/index.tsx [hunk 3]
```diff
@@ -23,6 +25,11 @@ export function Picker({
   keyId,
   userAddress,
   onChange,
+  collect = {
```
Comment (julien51) [inline unlock-app/src/components/interface/Picker/index.tsx:29]: Like!

**Review by julien51 — APPROVED** [review]

**Pull request merged**